code/modules/mob/new_player/new_player.dm

```diff
@@ -146,11 +146,12 @@
 
 				if(observer.client)
 					observer.client.change_view(world_view_size)
+					send_tacmap_assets_latejoin(observer)
```

Comment:
these snowflake calls really shouldn't be necessary, it should be totally viable to send these pngs when required through SSassets

Reply:
I need a suggestion on where you want these instead then. I can confirm it is necessary to send these assets for latejoiners - else they will have a wikimap as fallback because they don't have the current tacmap. Only until they've been present for a tacmap update will they then have a tacmap drawing thats on a flattened tactical map.

Both `tgui_interact` and `ui_static_data` are called every time the respective interface is opened by that user, so to not oversend the maps unnecessarily we'd have to either check a var stored on the mob or client, or look them up in some global list.

As they are, they are then only sent when needed. They currently are only unnecessarily sent if the player is sent back to lobby and they late join or observe again.

Reply:
Let me know if you meant something else by this.